src/test/java/org/beanpod/switchboard/controller/ChannelControllerTest.java

```diff
@@ -0,0 +1,160 @@
+      encoderDto = EncoderFixture.getEncoderDto();
+      decoderDto = DecoderFixture.getDecoderDto();
+      channelDto = ChannelFixture.getChannelDto();
+      inputChannelDto = ChannelFixture.getInputChannelDto();
+      outputChannelDto = ChannelFixture.getOutputChannelDto();
+   }
+
+   @BeforeEach
+   void setUp() {
+      initMocks(this);
+   }
+
+   @Test
+   void testRetrieveAllChannels() {
+      when(channelService.getChannels()).thenReturn(channelEntityList);
+      when(channelMapper.toChannelDTOs(any())).thenReturn(channelDtoList);
+      List<ChannelDTO> channelDTOS = channelController.retrieveAllChannels();
+      assertEquals(channelDtoList, channelDTOS);
+   }
+
+   @Test
+   void testRetrieveChannel() {
+      when(channelService.findChannel(ChannelFixture.CHANNEL_ID)).thenReturn(Optional.of(channelDto));
+      ResponseEntity<ChannelDTO> channelDTOResponseEntity = channelController.retrieveChannel(ChannelFixture.CHANNEL_ID);
+      assertEquals(channelDto, channelDTOResponseEntity.getBody());
+   }
+
+   @Test
+   void testRetrieveChannelEmpty() {
+      assertThrows(ExceptionType.DeviceNotFoundException.class, () -> {
+         channelController.retrieveChannel(ChannelFixture.CHANNEL_ID);
+      });
+   }
+
+   @Test
+   void createChannel() {
+      when(channelService.findChannel(ChannelFixture.CHANNEL_ID)).thenReturn(Optional.empty());
+      ResponseEntity<ChannelDTO> channel = channelController.createChannel(channelDto);
+      assertEquals(200, channel.getStatusCodeValue());
+   }
+
+   @Test
+   void createExistingChannel() {
+      when(channelService.findChannel(ChannelFixture.CHANNEL_ID)).thenReturn(Optional.of(channelDto));
+      assertThrows(ExceptionType.DeviceAlreadyExistsException.class, () -> {
+         channelController.createChannel(channelDto);
+      });
+   }
+
+   @Test
+   void createInputChannel() {
+      when(channelService.saveInputChannel(any())).thenReturn(inputChannelDto);
+      when(channelService.findChannel(ChannelFixture.CHANNEL_ID)).thenReturn(Optional.of(channelDto));
+      when(decoderService.findDecoder("1")).thenReturn(Optional.of(decoderDto));
+      ResponseEntity<InputChannelDTO> inputChannel = channelController.createInputChannel(ChannelFixture.CHANNEL_ID, "1");
+      assertEquals(inputChannelDto, inputChannel.getBody());
+   }
+
+   @Test
+   void createOutputChannel() {
+      when(channelService.saveOutputChannel(any())).thenReturn(outputChannelDto);
+      when(channelService.findChannel(ChannelFixture.CHANNEL_ID)).thenReturn(Optional.of(channelDto));
+      when(encoderService.findEncoder("1")).thenReturn(Optional.of(encoderDto));
+      ResponseEntity<OutputChannelDTO> outputChannel = channelController.createOutputChannel(ChannelFixture.CHANNEL_ID, "1");
+      assertEquals(outputChannelDto, outputChannel.getBody());
+   }
+
+   @Test
+   void deleteOutputChannel() {
+      when(channelService.deleteOutputChannelById(4569L)).thenReturn(1L);
```

Comment:
I think this `4569L` slipped through the cracks.

Reply:
cracks stopped leaking,fixed